pandas/io/excel/_base.py

```diff
@@ -1170,6 +1172,10 @@ def __init__(
         if engine is not None and engine not in self._engines:
             raise ValueError(f"Unknown engine: {engine}")
 
+        # First argument can also be bytes, so create a buffer
```

Comment:
is there a test that hits this?

Reply:
yes there are several, both failing and passing. E.g. `pandas/tests/io/excel/test_readers.py::TestReaders::test_corrupt_bytes_raises`